api/src/auth/drivers/oauth2.ts

```diff
@@ -123,10 +122,10 @@ export class OAuth2AuthDriver extends LocalAuthDriver {
 		}
 
 		const canRegisterUser =
-			!this.config.publicRegistrationFilter || validatePayload(this.config.publicRegistrationFilter, userInfo).length;
-
+			!this.config.publicRegistrationFilter || !validatePayload(this.config.publicRegistrationFilter, userInfo).length;
```

Comment:
Are you sure that `!validatePayload` is right?! I thought getting a length back from `validatePayload` would mean success, and `canRegisterUser` is truthy? 🤔

Reply:
> I thought getting a length back from validatePayload would mean success

`validatePayload` returns an array of validation errors (eg "field x has to be y"), so length = 0 is a good thing

Reply:
AHA! Well, that's pretty semantically screwy, but as long as it's correct! 😂

Reply:
validatePayload returns an array of errors, so when the length is 0 it is a success.